functions/packages.sh

```diff
@@ -414,7 +418,10 @@ influxdb_grafana_setup() {
     cond_redirect apt-key adv --keyserver keyserver.ubuntu.com --recv-keys 379CE192D401AB61 || FAILED=2
   else
     cond_redirect wget -O - https://packagecloud.io/gpg.key | apt-key add - || FAILED=2
-    echo "deb https://packagecloud.io/grafana/stable/debian/ $codename main" > /etc/apt/sources.list.d/grafana.list || FAILED=2
+    echo "deb https://packagecloud.io/grafana/stable/$distributor/ $codename main" > /etc/apt/sources.list.d/grafana.list || FAILED=2
+    if [ $distributor = "ubuntu" ]; then #remove this loop when grafana ubuntu packages are available
+        echo "deb https://packagecloud.io/grafana/stable/debian/ jessie main" > /etc/apt/sources.list.d/grafana.list || FAILED=2
+    fi
```

Comment:
Correct me if I am wrong but isn't something like this what we are after as there are no `ubuntu` packages available?

```
if [ $codename = "jessie" ]; then
    echo "deb https://packagecloud.io/grafana/stable/debian/ jessie main" > /etc/apt/sources.list.d/grafana.list || FAILED=2
else
    echo "deb https://packagecloud.io/grafana/stable/debian/ stretch main" > /etc/apt/sources.list.d/grafana.list || FAILED=2
fi
```

Reply:
Have not tested above code, could contain syntax error etc.

Reply:
well this is what my changes do, with the added benefit that once when there will be the packages for Ubuntu will be there, the if loop can be removed and we end up with a cleaner code with less points to adjust when a new supported stable release from, let's say from debian, is released.

But of course we can do it now withe the if-else loop, this should not be my decision

Reply:
Totally misread your if statement. It's brilliant and I like it a lot!

Reply:
But doesn't we end up with an invalid entry in `sources.list` for `Ubuntu` then which we can't resolve?
Eg. `"deb https://packagecloud.io/grafana/stable/ubuntu/ xenial  main"` will produces sources can't be found.

Reply:
well no ... line 421 would insert `deb https://packagecloud.io/grafana/stable/ubuntu/ xenial main` but than it will be overwritten by the if loop with `deb https://packagecloud.io/grafana/stable/debian/ jessie main´ 
i have planed this for the future, when there maybe will be ubuntu packages for grafana, then we can just remove the if loop and we will be done